GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -333,12 +366,115 @@ public void Clear()
             Invalidate(invalidateChildren: true);
         }
 
+        public void EndSelectionAtLoad()
+        {
+            _loadedToBeSelectedRevisionsCount = 0;
+            ToBeSelectedObjectIds = Array.Empty<ObjectId>();
+            _toBeSelectedGraphIndexes = null;
+        }
+
+        public void EnsureRowVisible(int row)
+        {
+            int countVisible = DisplayedRowCount(includePartialRow: false);
+            int firstVisible = FirstDisplayedScrollingRowIndex;
+            if (row >= 0 && (row < firstVisible || firstVisible + countVisible <= row))
+            {
+                FirstDisplayedScrollingRowIndex = row;
+            }
+        }
+
+        public ObjectId? GetFirstNotSelectedObjectId()
+        {
+            if (_loadedToBeSelectedRevisionsCount > 0 || ToBeSelectedObjectIds.Count == 0)
+            {
+                // At least one of the revisions was selected
+                return null;
+            }
+
+            return ToBeSelectedObjectIds[0];
+        }
+
+        public void ToBeSelectFirstFoundParent(IEnumerable<ObjectId> headParents)
+        {
+            if (headParents is not null)
+            {
+                foreach (var parentId in headParents)
+                {
+                    if (_revisionGraph.TryGetRowIndex(parentId, out int _))
+                    {
+                        ToBeSelectedObjectIds = new ObjectId[] { parentId };
+                        _loadedToBeSelectedRevisionsCount = ToBeSelectedObjectIds.Count;
+                        break;
+                    }
+                }
+            }
+        }
+
         public void LoadingCompleted()
         {
+            if (_loadedToBeSelectedRevisionsCount < ToBeSelectedObjectIds.Count)
+            {
+                // All expected revisions not found, settle with partial (empty) match
+                _loadedToBeSelectedRevisionsCount = ToBeSelectedObjectIds.Count;
+            }
+
+            if (_loadedToBeSelectedRevisionsCount > 0 && _revisionGraph.Count > 0)
+            {
+                // Rows have not been selected yet
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    int scrollTo = GetGraphIndexes().Count > 0 ? GetGraphIndexes().Max() : 0;
+                    await this.SwitchToMainThreadAsync();
+                    if (RowCount - 1 < scrollTo)
+                    {
+                        // Wait for the background thread to load all rows in the grid
+                        while (RowCount - 1 < scrollTo)
+                        {
+                            int maxScroll = Math.Min(RowCount - 1, scrollTo);
+                            EnsureRowVisible(maxScroll);
+
+                            UpdateVisibleRowRange();
+                            await Task.Delay(25);
```

Comment:
Why do we need this, and where did "25" come from?

Reply:
Added constant `backgroundPeriod` and updated the comment.
There is no point in polling more often than that.
(changed in tests too.)